src/main/java/seedu/command/DeleteTransactionCommand.java

```diff
@@ -0,0 +1,61 @@
+package seedu.command;
+
+import seedu.transaction.TransactionList;
+
+import java.util.List;
+
+public class DeleteTransactionCommand extends Command {
+    public static final String COMMAND_WORD = "delete-transaction";
+    public static final String COMMAND_GUIDE = "delete-transaction i/ INDEX: Delete a transaction";
+    public static final String[] COMMAND_MANDATORY_KEYWORDS = {"i/"};
+    public static final String[] COMMAND_EXTRA_KEYWORDS = {};
+
+    protected TransactionList transactions;
+
+    public DeleteTransactionCommand(TransactionList transactions) {
+        this.transactions = transactions;
+    }
+
+    @Override
+    public List<String> execute() throws Exception {
```

Comment:
Instead of throwing an exception, you may catch it and return a message list with the error message.